crawler/src/main/java/com/movie/recommender/crawler/service/MovieService.java

```diff
@@ -1,25 +1,192 @@
+            movieDetails.setCrew(credits.getCrew());
         });
+
+        // Load movie keywords
+        Optional<MovieKeywords> keywordsOpt = tmdbApiClient.getMovieKeywords(movieId.toString());
+        keywordsOpt.ifPresent(keywords -> movieDetails.setKeywords(keywords.getKeywords()));
+
+        return Optional.of(movieDetails);
+    }
+
+
+    public void loadShowtimes() {
+        Map<String, String> countryCodeMapping = Map.of(
+                "Ukraine", "ua"
+        );
+
+        List<NowPlayingMoviesByCountry> nowPlayingMovies = mongoDBService.getNowPlayingMovies();
+        List<CountryWithCitiesDTO> countries = locationServiceClient.getAllCountriesAndCities();
+        int apiCallCount = 0;
+
+        List<ShowtimesByCity> allShowtimes = new ArrayList<>();
+
+        // Iterate through each country and city
+        for (CountryWithCitiesDTO countryWithCities : countries) {
+            String countryName = countryWithCities.getCountryName();
+            String countryCode = countryCodeMapping.get(countryName);
+
+            if (countryCode == null) {
+                log.warn("Country code not found for country '{}'. Skipping...", countryName);
+                continue;
+            }
+
+            log.info("Loading showtimes for country '{}'", countryName);
+
+            for (String cityName : countryWithCities.getCities()) {
+                ShowtimesByCity showtimesByCity = new ShowtimesByCity();
+                showtimesByCity.setCountryCode(countryCode);
+                showtimesByCity.setCityName(cityName);
+
+                List<ShowtimesByCity.MovieIdentifier> movieList = new ArrayList<>();
+
+                // Iterate through now playing movies for the current country
+                for (NowPlayingMoviesByCountry nowPlayingMovie : nowPlayingMovies) {
+                    if (!nowPlayingMovie.getCountryCode().equals(countryCode)) {
+                        continue;
+                    }
+
+                    for (NowPlayingMoviesByCountry.MovieIdentifier movie : nowPlayingMovie.getResults()) {
+                        // Stop processing if the API call limit of 3 is reached
+
+                        ShowtimesByCity.MovieIdentifier showtimeMovie = new ShowtimesByCity.MovieIdentifier();
+                        showtimeMovie.setId(movie.getId());
+                        showtimeMovie.setTitle(movie.getTitle());
+
+                        // Fetch showtimes from the API
+                        Optional<MovieShowtimesResponse> showtimesResponse = serpApiClient.getMovieShowtimes(movie.getTitle(), cityName, "en");
+
+                        if (showtimesResponse.isPresent()) {
+                            showtimeMovie.setShowtimes(showtimesResponse.get().getShowtimes());
+                            movieList.add(showtimeMovie);
+                            apiCallCount++;
+                        }
+
+                        // Check API call limit again
+                        if (apiCallCount >= 3) {
```

Comment:
this logic is incorrect
you must break if the limit is reached
saving should be handled in other loop iterations or steps